pkg/agent/datamodel/defaults.go

```diff
@@ -389,20 +392,20 @@ func (cs *ContainerService) setOrchestratorDefaults(isUpgrade, isScale bool) {
 			// Distro assignment for masterProfile
 			if cs.Properties.MasterProfile.Distro == "" && cs.Properties.MasterProfile.ImageRef == nil {
 				if cs.Properties.OrchestratorProfile.IsKubernetes() && cs.Properties.OrchestratorProfile.KubernetesConfig.CustomHyperkubeImage == "" {
-					cs.Properties.MasterProfile.Distro = api.AKSUbuntu1604
+					cs.Properties.MasterProfile.Distro = AKSUbuntu1604
 				} else {
-					cs.Properties.MasterProfile.Distro = api.Ubuntu
+					cs.Properties.MasterProfile.Distro = Ubuntu
 				}
 			} else if cs.Properties.OrchestratorProfile.IsKubernetes() && (isUpgrade || isScale) {
-				if cs.Properties.MasterProfile.Distro == api.AKSDockerEngine || cs.Properties.MasterProfile.Distro == api.AKS1604Deprecated {
-					cs.Properties.MasterProfile.Distro = api.AKSUbuntu1604
-				} else if cs.Properties.MasterProfile.Distro == api.AKS1804Deprecated {
-					cs.Properties.MasterProfile.Distro = api.AKSUbuntu1804
+				if cs.Properties.MasterProfile.Distro == AKSDockerEngine || cs.Properties.MasterProfile.Distro == AKS1604Deprecated {
+					cs.Properties.MasterProfile.Distro = AKSUbuntu1604
+				} else if cs.Properties.MasterProfile.Distro == AKS1804Deprecated {
+					cs.Properties.MasterProfile.Distro = AKSUbuntu1804
 				}
 			}
 			// The AKS Distro is not available in Azure German Cloud.
 			if cloudSpecConfig.CloudName == api.AzureGermanCloud {
-				cs.Properties.MasterProfile.Distro = api.Ubuntu
+				cs.Properties.MasterProfile.Distro = Ubuntu
```

Comment:
and this, no need to maintain masterProfile

Reply:
There are many code that uses "MasterProfile" that we don't need. I'll send a separate PR to remove them together.